### DISCUSSION TITLE
How can i help?

### DISCUSSION THREAD

**Discussion by prashant235913** [AppFlowy-IO/AppFlowy#2502] [General]
Hey!This is Prashant Mishra.A learner of DevOps tech from India.Currently I am learning about Kubernetes and Docker . Is there any way That I can contribute to this project.

**Comment by bartvdbraak** [comment]
Hello @prashant235913,

Thank you for showing interest in contributing to the AppFlowy project. It's great to have you on board! Here's a step-by-step guide to help you get started:

1. Familiarize Yourself with the Project:
   - Take some time to explore the repository, its structure, and the codebase.
   - Read the README file thoroughly to understand the project's goals, guidelines, and contribution process.
   - Review the open issues and pull requests to get an idea of ongoing discussions and areas that require attention.

2. Choose an Issue to Work On:
   - Visit the "Issues" tab in the repository to find a list of open issues.
   - Look for issues that align with your skills, interests, and expertise in Kubernetes and Docker.
   - Feel free to ask for clarification or additional information on any issue you're interested in.

3. Fork the Repository:
   - Click on the "Fork" button at the top right corner of the repository page to create your own copy of the project.

4. Set Up Your Local Environment:
   - Clone the forked repository to your local machine using the Git command: `git clone https://github.com/your-username/AppFlowy.git`
   - Install any necessary dependencies mentioned in the README or documentation.

5. Create a New Branch:
   - Before making any changes, create a new branch to work on your contribution using the Git command: `git checkout -b my-contribution`
   - Give the branch an informative name that reflects the nature of your work.

6. Make the Changes:
   - Implement the necessary changes or additions based on the selected issue.
   - Follow the project's coding style guidelines and best practices.
   - Regularly commit your changes with descriptive commit messages using the Git command: `git commit -m "Your message"`

7. Test Your Changes:
   - Ensure that your changes are working as intended.
   - Write tests if appropriate and make sure existing tests pass.

8. Submit a Pull Request (PR):
   - Push your changes to your forked repository: `git push origin my-contribution`
   - Visit the original AppFlowy repository and click on the "Pull Request" button.
   - Provide a clear and concise title and description for your PR, referencing the issue it addresses.
   - Submit the PR and wait for feedback or requests for changes.

9. Engage in the Discussion:
   - Be responsive to any feedback, comments, or suggestions from the maintainers or other contributors.
   - Discuss any necessary modifications or improvements to your code.

10. Celebrate Your Contribution:
    - Once your PR is accepted and merged, congratulations! You're an official contributor to AppFlowy.
    - Get in touch with the project maintainers, usually via the provided contact link, to receive your special Contributor T-shirt.
    - Wear your T-shirt proudly and share it by tagging @appflowy on Twitter to showcase your accomplishment.

Remember, contributions of all sizes are valuable, so don't hesitate to ask questions, collaborate with others, and have fun while learning and improving the project.

If you need further guidance or have any specific questions, feel free to ask. Happy contributing!

**Comment by prashant235913** [comment]
Thanks @bartvdbraak ,Doubts are clear

**Reply by bartvdbraak** [reply]
Great to hear! 

If you found my reply useful, please consider to accept it as an answer!

**Reply by prashant235913** [reply]
Thanks again @bartvdbraak , I just made a quick Tutorial about an open source contribution.You can share it with anyone who's just starting with Open source.Thanks again.
Video link:-
 
https://youtu.be/C4s1CtswKbA
Also tell me how can I make your comment as an Answer,I am unaware of this.

**Reply by bartvdbraak** [reply]
Hi @prashant235913, 

I watched your video and it's great for giving new people a look into how to contribute to open source projects!

Also, my apologies, I thought this discussion thread was opened as a question, but this thread is labeled as a **conversation**. So don't worry about marking it as an answer, that can only happen in opened threads in these sections apparently:
- [General help wanted](https://github.com/AppFlowy-IO/AppFlowy/discussions/categories/general-help-wanted)
- [Technical help wanted](https://github.com/AppFlowy-IO/AppFlowy/discussions/categories/technical-help-wanted)

**Comment by prashant235913** [comment]
@bartvdbraak Thanku for the appreciation , it means a lot.
Also thanks for reminding me of those seperate threads for seperate types of questions.
